metricshub-engine/src/test/java/org/sentrysoftware/metricshub/engine/strategy/discovery/DiscoveryStrategyTest.java

```diff
@@ -208,5 +209,34 @@ void testRun() throws Exception {
 			PHYSICAL_DISK.getKey()
 		);
 		assertEquals(expectedOrder, discoveredMonitors.keySet());
+
+		// Check that StatusInformation is collected on the connector monitor (criterion processing success case)
+		assertEquals(
+			"Received Result: 1.3.6.1.4.1.795.10.1.1.3.1.1.0\tASN_OCTET_STR\tTest. SnmpGetNextCriterion test succeeded:\n" +
+			"SnmpGetNextCriterion(super=SnmpCriterion(super=Criterion(type=snmpGetNext, forceSerialization=false), oid=1.3.6.1.4.1.795.10.1.1.3.1.1, expectedResult=null))\n" +
```

Comment:
Can you please @MedMaalej update the toString() method on `DeviceTypeCriterion`, `HttpCriterion`, `ProcessCriterion`, `ProductRequirementsCriterion` and `SnmpCriterion`. Please use the same pattern as on `CommandLineCriterion`. Thanks.